### ISSUE TITLE
Background update crash

### ISSUE THREAD

**Issue by StoyanDimitrov** [Tobi823/ffupdater#21]
A device is with disabled mobile data but with Wi-Fi turned on. Approximately an hour after the device lose (Wi-Fi) connection because it is out of range an error of the sort appears:
```
Caused by: java.net.UnknownHostException: Unable to resolve host "firefox-ci-tc.services.mozilla.com": No address associated with hostname
```
<details>
<summary>Full stacktrace</summary>
<pre>de.marmaro.krt.ffupdater.utils.ParamRuntimeException: Failed to fetch the download url from Mozilla CI
	at de.marmaro.krt.ffupdater.InstallActivity.lambda$fetchAvailableMetadata$3$InstallActivity(InstallActivity.java:207)
	at de.marmaro.krt.ffupdater.-$$Lambda$InstallActivity$D3MDEMe3EyG_NptgNd2sctku1ks.run(Unknown Source:4)
	at java.lang.Thread.run(Thread.java:919)
Caused by: java.util.concurrent.ExecutionException: de.marmaro.krt.ffupdater.utils.ParamRuntimeException: can't consume API interface https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.latest.arm64-v8a/artifacts/public/chain-of-trust.json
	at java.util.concurrent.FutureTask.report(FutureTask.java:123)
	at java.util.concurrent.FutureTask.get(FutureTask.java:207)
	at de.marmaro.krt.ffupdater.InstallActivity.lambda$fetchAvailableMetadata$3$InstallActivity(InstallActivity.java:200)
	... 2 more
Caused by: de.marmaro.krt.ffupdater.utils.ParamRuntimeException: can't consume API interface https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.latest.arm64-v8a/artifacts/public/chain-of-trust.json
	at de.marmaro.krt.ffupdater.metadata.fetcher.ApiConsumer.consume(ApiConsumer.java:43)
	at de.marmaro.krt.ffupdater.metadata.fetcher.MozillaCiConsumer.consume(MozillaCiConsumer.java:22)
	at de.marmaro.krt.ffupdater.metadata.fetcher.Firefox.call(Firefox.java:47)
	at de.marmaro.krt.ffupdater.metadata.fetcher.Fetcher.fetchAvailableMetadata(Fetcher.java:80)
	at de.marmaro.krt.ffupdater.metadata.fetcher.Fetcher.call(Fetcher.java:56)
	at de.marmaro.krt.ffupdater.metadata.fetcher.Fetcher.call(Fetcher.java:15)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	... 1 more
Caused by: java.net.UnknownHostException: Unable to resolve host "firefox-ci-tc.services.mozilla.com": No address associated with hostname
	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:156)
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:103)
	at java.net.InetAddress.getAllByName(InetAddress.java:1152)
	at com.android.okhttp.Dns$1.lookup(Dns.java:41)
	at com.android.okhttp.internal.http.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:178)
	at com.android.okhttp.internal.http.RouteSelector.nextProxy(RouteSelector.java:144)
	at com.android.okhttp.internal.http.RouteSelector.next(RouteSelector.java:86)
	at com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:176)
	at com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:128)
	at com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:97)
	at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:289)
	at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:232)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:465)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:411)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getHeaders(HttpURLConnectionImpl.java:167)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getHeaderField(HttpURLConnectionImpl.java:211)
	at java.net.URLConnection.getContentEncoding(URLConnection.java:529)
	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getContentEncoding(DelegatingHttpsURLConnection.java:143)
	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getContentEncoding(HttpsURLConnectionImpl.java:30)
	at de.marmaro.krt.ffupdater.metadata.fetcher.ApiConsumer.consume(ApiConsumer.java:33)
	... 9 more
Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
	at libcore.io.Linux.android_getaddrinfo(Native Method)
	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:74)
	at libcore.io.BlockGuardOs.android_getaddrinfo(BlockGuardOs.java:200)
	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:74)
	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:135)
	... 28 more</pre>
</details>

It is reproducible. Are you interested in the exact steps? Probably related to #18

**Comment by Tobi823** [comment]
Yes, it is related to #18.

How often does this error occur? Once a week?

**Comment by StoyanDimitrov** [comment]
Every time I go out of Wi-Fi reach. And it is not immediate

**Comment by Tobi823** [comment]
The issue should now be fixed (in version 71.0.0).

FFUpdater retries up to two times before crashing (when contacting the server for available versions)
